### PR TITLE
build gir objects when cross compiling

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by tschoonj** [comment]
Why is this necessary? In build.sh `-Dgir=true` is passed to meson so they should get built already.

**Comment by isuruf** [comment]
They aren't. In the logs for master it says `introspection: false`

**Comment by tschoonj** [comment]
Looks like they renamed the option recently. Can you drop `-Dgir=true` in build.sh and bld.bat, and add the new option in the bld.bat as well? Thanks!

**Comment by tschoonj** [comment]
Thanks!

**Comment by tschoonj** [comment]
Why does the build need to go into `BUILD_PREFIX` for the OSX ARM build?

**Comment by isuruf** [comment]
We are cross compiling from `osx-64` to `osx-arm64`. `gobject-introspection` machinery isn't really intended for cross compiling. `gobject-introspection` compiles a binary with the `osx-arm64` library and tries to run it which fails because we can't run `osx-arm64` binaries. So what I do is I create a native binary for `osx-64` and save the introspection files and then in the `osx-arm64` build instead of running the introspection binary, I load the introspection files from `osx-64`. This assumes that the introspection files for the two platforms `osx-64` and `osx-arm64` will be the same.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,6 +30,7 @@ requirements:
     - libpng    # [build_platform != target_platform]
     - zlib      # [build_platform != target_platform]
   host:
+    - gobject-introspection
```
Comment (tschoonj) [inline recipe/meta.yaml:33]: This should not be necessary, and should be avoided at all cost to prevent conflicts between meson and gobject-introspection.

Comment (isuruf) [reply recipe/meta.yaml:33]: It is needed in cross compiling, otherwise pkg-config will not find it. Not sure why

Comment (tschoonj) [reply recipe/meta.yaml:33]: In that case please add `# [build_platform != target_platform]` to it. Thanks

File: recipe/build.sh [hunk 2]
```diff
@@ -11,47 +11,73 @@ if [[ "$(uname)" = Darwin ]] ; then
...
-if [[ "$target_platform" == "osx-arm64" && "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
-    echo "[host_machine]" > cross_file.txt
-    echo "system = 'darwin'" >> cross_file.txt
-    echo "cpu_family = 'aarch64'" >> cross_file.txt
-    echo "cpu = 'arm64'" >> cross_file.txt
-    echo "endian = 'little'" >> cross_file.txt
-    meson_options+=(--cross-file cross_file.txt)
-    export PKG_CONFIG=$BUILD_PREFIX/bin/pkg-config
+export PKG_CONFIG="$BUILD_PREFIX/bin/pkg-config"
+export PKG_CONFIG_PATH_FOR_BUILD="$BUILD_PREFIX/lib/pkgconfig"
```
Comment (tschoonj) [inline recipe/build.sh:74]: Are `PKG_CONFIG `and  `PKG_CONFIG_PATH_FOR_BUILD` used somewhere?

Comment (isuruf) [reply recipe/build.sh:74]: yes, they are used by meson

**Review by tschoonj — APPROVED** [review]
Looks good... even though I really hope a native build on macOS ARM will be possible sometime soon so things may be simplified...

**Comment by isuruf** [comment]
Thanks for the review

**Pull request merged**